go-controller/pkg/ovn/pods.go

```diff
@@ -682,9 +676,24 @@ func (oc *Controller) ovnNBLSPDel(logicalPort, logicalSwitch, lspUUID string) ([
 	if lspUUID == "" {
 		ctx, cancel := context.WithTimeout(context.Background(), ovntypes.OVSDBTimeout)
 		defer cancel()
-		if err := oc.nbClient.Get(ctx, lsp); err != nil {
+		if err := oc.nbClient.Get(ctx, lsp); err != nil && err != libovsdbclient.ErrNotFound {
 			return nil, fmt.Errorf("cannot delete logical switch port %s failed retrieving the object %v", logicalPort, err)
+		} else if err == libovsdbclient.ErrNotFound {
+			// lsp doesn't exist; nothing to do
+			return allOps, nil
+		}
+	}
+
+	ls := &nbdb.LogicalSwitch{}
+	var err error
+	if lsUUID, ok := oc.lsManager.GetUUID(logicalSwitch); !ok {
+		klog.Errorf("Error getting logical switch for node %s: switch not in logical switch cache", logicalSwitch)
+		// Not in cache: Try getting the logical switch from ovn database (slower method)
+		if ls, err = libovsdbops.FindSwitchByName(oc.nbClient, logicalSwitch); err != nil {
+			return nil, fmt.Errorf("can't find switch for node %s: %v", logicalSwitch, err)
 		}
+	} else {
```

Comment:
Is the else required?

Reply:
Yes right? Only if the entry exists in cache we fill the UUID for the switch, else we take another path.

Reply:
Thanks for the clarification @tssurya . I still have a question, if I get correctly (probably not).

So, we will start with `oc.lsManager.GetUUID()` , it will go to: https://github.com/ovn-org/ovn-kubernetes/blob/779b9cdb10edb3e06b1eea73fa469251e42cc56b/go-controller/pkg/ovn/logical_switch_manager/logical_switch_manager.go#L40:38  and return with `true or false`. Lets assume `false` and go to `if ls, err = libovsdbops.FindSwitchByName(oc.nbClient, logicalSwitch); err != nil {` path.

What happens if  `FindSwitchByName` returns ok? we will have the object in `ls`, if not, return `fmt.Errorf`  correct?

On the other hand, if `oc.lsManager.GetUUID()` returns `ok`, we will have in `lsUUID`. 

Not clear yet about the else. I might need to setup a debug session here.

Reply:
> Thanks for the clarification @tssurya . I still have a question, if I get correctly (probably not).
> 
> So, we will start with `oc.lsManager.GetUUID()` , it will go to: https://github.com/ovn-org/ovn-kubernetes/blob/779b9cdb10edb3e06b1eea73fa469251e42cc56b/go-controller/pkg/ovn/logical_switch_manager/logical_switch_manager.go#L40:38 and return with `true or false`. Lets assume `false` and go to `if ls, err = libovsdbops.FindSwitchByName(oc.nbClient, logicalSwitch); err != nil {` path.
> 
> What happens if `FindSwitchByName` returns ok? we will have the object in `ls`, if not, return `fmt.Errorf` correct?
> 
> On the other hand, if `oc.lsManager.GetUUID()` returns `ok`, we will have in `lsUUID`.
> 
> Not clear yet about the else. I might need to setup a debug session here.

what you said is correct ^ we utilize the switch manager cache for efficiency first and see if UUID already exists, if it doesn't we go to DB to fetch the object and use that if ls is returned.